SpatialGDK/Source/SpatialGDKTests/Examples/SpatialGDKExampleTest.cpp

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "TestDefinitions.h"
+
+#include "Misc/ScopeTryLock.h"
+#include "HAL/PlatformFilemanager.h"
+
+#define EXAMPLE_SIMPLE_TEST(TestName) \
+	TEST(SpatialGDKExamples, SimpleExamples, TestName)
+
+#define EXAMPLE_COMPLEX_TEST(TestName) \
+	COMPLEX_TEST(SpatialGDKExamples, ComplexExamples, TestName)
+
+DECLARE_LOG_CATEGORY_EXTERN(LogSpatialGDKExamples, Log, All);
+DEFINE_LOG_CATEGORY(LogSpatialGDKExamples);
+
+// 1. Latent command example
+namespace
+{
+	const double MAX_WAIT_TIME_FOR_SLOW_COMPUTATION = 2.0;
+	const double MIN_WAIT_TIME_FOR_SLOW_COMPUTATION = 1.0;
+	const double COMPUTATION_DURATION = 1.0;
+
+	struct
+	{
+		FCriticalSection Mutex;
+		int Value;
+	} ComputationResult;
+}
+
+DEFINE_LATENT_COMMAND(StartSlowComputation)
+{
+	AsyncTask(ENamedThreads::AnyBackgroundThreadNormalTask, []
+	{
+		FScopeLock SlowComputationLock(&ComputationResult.Mutex);
+		FPlatformProcess::Sleep(COMPUTATION_DURATION);
+		ComputationResult.Value = 42;
+	});
+
+	return true;
+}
+
+DEFINE_LATENT_COMMAND_ONE_PARAMETER(WaitForComputationAndCheckResult, FAutomationTestBase*, Test)
+{
+	const double TimePassed = FPlatformTime::Seconds() - StartTime;
+
+	if (TimePassed >= MIN_WAIT_TIME_FOR_SLOW_COMPUTATION)
+	{
+		FScopeTryLock SlowComputationLock(&ComputationResult.Mutex);
+
+		if (SlowComputationLock.IsLocked())
+		{
+			Test->TestTrue("", ComputationResult.Value == 42);
+			return true;
+		}
+
+		if (TimePassed >= MAX_WAIT_TIME_FOR_SLOW_COMPUTATION)
+		{
+			UE_LOG(LogSpatialGDKExamples, Error, TEXT("Computation timed out"));
+			return true;
+		}
+	}
+
+	return false;
+}
+
+EXAMPLE_SIMPLE_TEST(GIVEN_initial_value_WHEN_performing_slow_compuation_THEN_the_result_is_correct)
```

Comment:
Again this name doesn't really seem to work. The result should always be correct.
I don't know how much we care about this, but if these are meant to be exemplar tests then we might want this to work as well as possible.

Reply:
I'll rename it to `GIVEN_initial_value_WHEN_performing_computation_in_background_thread_THEN_the_result_is_correct`

Reply:
Why not come up with some simple test that does specify something though. Then we can name it as we'd name another test. If we have this as an example then there will likely be tests written that copy it.

Reply:
Could you elaborate?

Reply:
Discussed offline, I'll leave it as is for now.